cmd/monaco/deploy/deploy.go

```diff
@@ -123,39 +125,120 @@ func loadProjects(fs afero.Fs, manifestPath string, man *manifest.Manifest, spec
 	return projects, nil
 }
 
-func checkEnvironments(projects []project.Project, envs manifest.Environments) error {
+type KindCoordinates map[string][]coordinate.Coordinate
+type KindCoordinatesPerEnvironment map[string]KindCoordinates
+type CoordinatesPerEnvironment map[string][]coordinate.Coordinate
+
+func validateProjectsWithEnvironments(projects []project.Project, envs manifest.Environments) error {
```

Comment:
I would suggest making this as 2 step validation (split into 2 separate function). One to check environments, another to check projects.